core/types/src/attributes.rs

```diff
@@ -9,10 +9,47 @@ new_type_wrapper! {
     Attributes, sgx_attributes_t;
 }
 
+impl Default for Attributes {
+    fn default() -> Self {
+        Attributes(sgx_attributes_t { flags: 0, xfrm: 0 })
+    }
+}
+
+impl Attributes {
+    /// Set the `flags` for the attributes
+    ///
+    /// # Arguments
+    ///
+    /// * `flags` - The flags to be set in the attributes
+    pub fn set_flags(mut self, flags: u64) -> Self {
+        self.0.flags = flags;
+        self
+    }
+
+    /// Set the `transform` for the attributes
+    ///
+    /// # Arguments
+    ///
+    /// * `transmform` - The transform to be set to the `xfrm` in the attributes
+    pub fn set_transform(mut self, transform: u64) -> Self {
+        self.0.xfrm = transform;
+        self
+    }
+}
+
 new_type_wrapper! {
     MiscellaneousSelect, sgx_misc_select_t;
 }
 
+// Suppress clippy as the `new_type_wrapper` macro can't derive Default for many
+// of the types.
+#[allow(clippy::derivable_impls)]
```

Comment:
I'd honestly suggest changing this macro to not declare the struct so you can just do the manual derives. I don't know if this is still a thing, but CLion used to have a lot of difficulty figuring out that a type exists if it's defined in a macro...

Reply:
pulled declaration outside of macro